lib/tasks/port_legacy_products.rake

```diff
@@ -0,0 +1,7 @@
+namespace :custom_tasks do
+  task :port_legacy_products => :environment do
+    Product.all.each do |product|
+      product.update_column(:category_id, Category.first.id) if product.category.nil?
```

Comment:
Category.first.id will fire a query for each product. Take that out of the each loop. Do not use update_column. we want all the callbacks to run. 
Use find_each.
instead of `if` condition, you can use something like this
`product.category_id ||= first_category`